config/profile.go

```diff
@@ -809,11 +809,11 @@ func (p *Profile) SchedulableCommands() (commands []string) {
 	return
 }
 
-// Schedules returns a slice of ScheduleConfig that satisfy the schedule.Config interface
-func (p *Profile) Schedules() []*ScheduleConfig {
+// Schedules returns a slice of Schedule
+func (p *Profile) Schedules() []*Schedule {
```

Comment:
I'm struggling to find a nice _common_ way of passing parameters to a `Init` method.

Right now, we need a `*ScheduleBaseSection` which is coming from a scheduled command, along with its profile so we can initialise the environment variables.
For version 2, we'll need the new schedule struct which is going to be different than `ScheduleBaseSection` (along with its profile).

I guess we can add a common interface to `ScheduleBaseSection` and the new struct for `v2`? they should have similar fields.

Reply:
I think the Init you created is just fine and should work in either way. Not sure if we need more for the moment (besides adding env capture part, which I slightly modified in #323). For later it has all information needed to setup the `Schedule` instance, I think. 

The `profiles...` instance list in the args is just to avoid the need to load the profiles from the config if the instances exist already, it is however optional.

For the env capture with multiple profiles we may collect the first profile only, since the important part is the OS environment and that way the behaviour can be documented and will be more deterministic as if we would merge multiple profiles.

Reply:
.. or for the `Schedule.Environment`, it may make sense to not include profile specific environment at all since it is set by config anyway.

Reply:
Yes I think it can be left as a separate PR for now:
- you're going to change the implementation slightly in #323
- it's likely to change with the well defined requirements of v2

I'm going to merge it now, it's been working fine on some of my machines (all configs v1 though, I need to change a few configs to v2)

Sorry it ended up being quite big again, we're going to have some work on merging the other PRs 😆